boris-http/src/Boris/Http/Api/Build.hs

```diff
@@ -151,14 +153,48 @@ byProject store project =
 logOf :: Store -> LogService -> BuildId -> EitherT Store.FetchError IO (Maybe (Source IO ByteString))
 logOf store logs i = do
   d <- Store.fetch store i
-  case d >>= buildDataLog of
+  case d of
     Nothing ->
       pure Nothing
     Just ld -> do
       case logs of
-        CloudWatchLogs env ->
-          pure . Just $ source env ld =$= Conduit.map (\t ->
+        CloudWatchLogs env environ -> do
+          let
+            gname = LogGroup $ mconcat [ "boris.", renderEnvironment environ ]
+            sname = LogStream $ mconcat [ renderProject $ buildDataProject ld, ".", renderBuildId $ buildDataId ld ]
+          pure . Just $ source env (CloudwatchLogData gname sname) =$= Conduit.map (\t ->
             Text.encodeUtf8 $ t <> "\n")
+        DBLogs -> do
+          l' <- Store.fetchLogData store i
+          case l' of
+            Just (DBLog l) ->
+              pure . Just $ Conduit.sourceList l =$= Conduit.map (\t -> Text.encodeUtf8 $ renderDBLogData t <> "\n")
+            Just (CloudwatchLog _) ->
+              pure Nothing
+            Nothing ->
+              pure Nothing
+        DevNull ->
+          pure Nothing
+
+logOf' :: Store -> LogService -> BuildId -> EitherT Store.FetchError IO (Maybe [Text])
+logOf' store logs i = do
+  d <- Store.fetch store i
+  case d of
+    Nothing ->
+      pure Nothing
+    Just ld -> do
+      case logs of
+        CloudWatchLogs env environ ->
+          pure Nothing
```

Comment:
So this would do basically what logOf does, except it would run the conduit to get the final string, using something like runConduit $ <other code> =$= Data.Conduit.List.foldMap (\x -> x <> "\n") to get it back to a string.